src/MCPClient/lib/clientScripts/assignUUIDsToDirectories.py

```diff
@@ -0,0 +1,141 @@
+
+import argparse
+from functools import wraps
+import os
+import sys
+
+import django
+django.setup()
+# dashboard
+from main.models import Transfer
+# archivematicaCommon
+from custom_handlers import get_script_logger
+from databaseFunctions import create_directory_models
+from archivematicaFunctions import get_dir_uuids, format_subdir_path
+
+logger = get_script_logger('archivematica.mcp.client.assignUUIDsToDirectories')
+
+
+class DirsUUIDsException(Exception):
+    """If I am raised, return 1."""
+    exit_code = 1
+
+
+class DirsUUIDsWarning(Exception):
+    """If I am raised, return 0."""
+    exit_code = 0
+
+
+def exit_on_known_exception(func):
+    """Decorator that makes this module's ``main`` function cleaner by handling
+    early exiting by catching particular exceptions.
+    """
+    @wraps(func)
+    def wrapped(*_args, **kwargs):
+        try:
+            func(*_args, **kwargs)
+        except (DirsUUIDsException, DirsUUIDsWarning) as exc:
+            return exc.exit_code
+    return wrapped
+
+
+def _exit_if_not_include_dirs(include_dirs):
+    """Quit processing if include_dirs is not truthy."""
+    if not include_dirs:
+        logger.info('Configuration indicates that directories in this Transfer'
+                    ' should not be given UUIDs.')
+        raise DirsUUIDsWarning
+
+
+def _get_transfer_mdl(transfer_uuid):
+    """Get the ``Transfer`` model with UUID ``transfer_uuid``. Also update it
+    in the db to indicate that this transfer has UUIDs assigned to the
+    directories that it contains.
+    """
+    try:
+        transfer_mdl = Transfer.objects.get(uuid=transfer_uuid)
+        transfer_mdl.diruuids = True
+        transfer_mdl.save()
+        return transfer_mdl
+    except Transfer.DoesNotExist:
+        logger.warning('There is no transfer with UUID %s', transfer_uuid)
+        raise DirsUUIDsException
+
+
+def _get_subdir_paths(root_path):
+    """Return a generator of subdirectory paths in ``root_path``."""
+    if not os.path.isdir(root_path):
+        logger.warning('Transfer path %s is not a path to a directory',
+                       root_path)
+        raise DirsUUIDsException
+    # objects/ and root dirs need no UUIDs.
+    exclude_paths = (root_path, os.path.join(root_path, 'objects'))
+    return (format_subdir_path(dir_path, root_path) for dir_path, __, ___ in
+            os.walk(root_path) if dir_path not in exclude_paths)
+
+
+def str2bool(val):
+    """'True' is ``True``; aught else is ``False."""
+    if val == 'True':
+        return True
+    return False
+
+
+@exit_on_known_exception
+def main(transfer_path, transfer_uuid, include_dirs):
+    """Assign UUIDs to all of the directories (and subdirectories, i.e., all
+    unique directory paths) in the absolute system path ``transfer_path``, such
+    being the root directory of the transfer with UUID ``transfer_uuid``. Do
+    this only if ``include_dirs`` is ``True``.
+    """
+    _exit_if_not_include_dirs(include_dirs)
```

Comment:
I'm not sure I agree that the name doesn't describe the behaviour: it clearly states that the function should exit (i.e., return) under a certain condition. Just because Python doesn't succinctly let you return from function `f` by calling function `g` from `f`, doesn't mean you shouldn't be able to get around that. Note that the calls to `_get_subdir_paths` and `_get_transfer_mdl` are doing the same thing: they will cause `main` to exit if a known exception is raised too.

Would it help if I renamed `_exit_if_not_include_dirs` to `_return_if_not_include_dirs` or `_assert_include_dirs`? Or do you really just want to see `return` statements in `main`?